### PR TITLE
/login flow, sessions, bcryptjs encryption module, modified bodyParser for POST requests

### PR DESCRIPTION
**Pull request by toucan-stan**
Added a basic /login flow, added support for express-session, added the bcryptjs module to packages.json for password encryption/hashing (to be implemented later), and modified bodyParser to support basic HTTP POST requests.

### CODE REVIEW

**Comment by nathanperkins** [comment]
Looks great!

What do you think of achieving a few of these things before we pull this one in? I can help with these :)

1. [Make user.email field unique](http://mongoosejs.com/docs/2.7.x/docs/indexes.html) since we're using it as a login.
1. Finish up the registration route and view.
1. Refactor the login logic into a static helper function on the UserModel or maybe in a separate module (/helpers/auth?)
1. [Modify the UserModel pre(save) logic](http://devsmash.com/blog/password-authentication-with-mongoose-and-bcrypt) so that hashing the password is handled by the model.
1. Add some UserModel tests to confirm the login logic.

Unique Field Example
```
var User = new Schema({
    name: { type: String, unique: true }
})
```

File: models/UserModel.js [hunk 1]
```diff
@@ -1,17 +1,92 @@
-const mongoose = require('mongoose');
-const Schema = mongoose.Schema;
+const mongoose = require('mongoose'), 
+      bcrypt = require('bcryptjs'),
+      Schema = mongoose.Schema,
+      SALT_WORK_FACTOR = 10;
 
-const userSchema = mongoose.Schema({
+const UserSchema = mongoose.Schema({
     firstName: {type: String, required: true},
```
Comment (sarahmaas) [inline models/UserModel.js]: Unique was one of the changes I was talking about wanting to add earlier. Thanks! :) Please add trim: true to firstName, lastName, and email.

Comment (nathanperkins) [reply models/UserModel.js]: done :)

File: app.js [hunk 2]
```diff
@@ -21,8 +22,10 @@ const ip = settings.ip || "localhost";
 const appName = settings.app_name || "homework-bet-dev";
 
 app.set('view engine', 'ejs');
-app.use(express.static('public'))
+app.use(express.static('public'));
+app.use(bodyParser.urlencoded( {extended: false} ));
 app.use(bodyParser.json());
+app.use(session({secret: 'bobross'}));
```
Comment (nathanperkins) [inline app.js]: lets move the session secret out into the settings file so that it's not hardcoded

File: routes/_routes.js [hunk 3]
```diff
@@ -16,17 +19,96 @@ router.get('/', function (req, res) {
...
 router.get('/register', function (req, res) {
-    res.redirect('/users/new');
+    res.render('register', {
+        appName: appName,
+        pageTitle: "Register",
+    });
+});
+ 
+
+router.post('/register', function(req, res) {
```
Comment (nathanperkins) [inline routes/_routes.js:35]: RESTful route would be POST /users, so we would probably move this to the users route

File: routes/_routes.js [hunk 4]
```diff
@@ -16,17 +19,96 @@ router.get('/', function (req, res) {
...
+
+    var user = new User({
+        firstName: req.body.firstName,
+        lastName: req.body.lastName,
+        password: req.body.password, 
+        email: req.body.email,
+    });
+
+    user.save( function(err, data) {
```
Comment (nathanperkins) [inline routes/_routes.js:44]: I'll change this to use promises

File: routes/_routes.js (hunk 3 above)
Comment (nathanperkins) [reply routes/_routes.js:35]: we will do this later

**Review by nathanperkins — DISMISSED** [review]

**Comment by nathanperkins** [comment]
I'll fix these warnings:
```
express-session deprecated undefined resave option; provide resave option app.js:28:9
express-session deprecated undefined saveUninitialized option; provide saveUninitialized option app.js:28:9
```

**Pull request merged**